src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeEventProcessorTest.java

```diff
@@ -123,24 +134,43 @@ public void should_dead_letter_the_message_when_unrecoverable_failure() throws E
 
     @Test
     public void should_not_finalize_the_message_when_recoverable_failure() throws Exception {
-        // given
-        reset(eventPublisherContainer);
-        willThrow(new RuntimeException("test exception"))
-            .given(eventPublisherContainer)
-            .getPublisher(any(), any());
+        verifyNoInteractionsWhenMessageProcessingFails(messageOperations);
+    }
 
+    @Test
+    public void should_not_do_anything_in_notify_exception() {
+        // when
+        processor.notifyException(null, null);
+    }
+
+    @Test
+    public void should_send_message_with_envelope_id_when_processing_successful() throws Exception {
         // when
         CompletableFuture<Void> result = processor.onMessageAsync(someMessage);
         result.join();
 
         // then
-        verifyNoMoreInteractions(messageOperations);
+        verify(processedEnvelopeNotifier).notify(ENVELOPE_ID);
```

Comment:
what's `someMessage` and how do I know `ENVELOPE_ID` is related to it?

Reply:
perhaps you could retrieve expected envelope id from the message used in the 'when' stage?